### ISSUE TITLE
Code Push App Size Limit

### ISSUE THREAD

**Issue by bmanderscheid** [Microsoft/code-push#291]
I'm seriously considering using Code Push for an iPad application we build / maintain for our client.  I can't find documentation on the limitations, most importantly app file size limits.  Can someone shed light on this matter? Thanks!

**Comment by silhouettes** [comment]
The limit is 200 MB at the moment. Out of curiosity, how big is your app?

Note that each time you perform a release, you'll need to upload the whole app, and the server will take care of generating file-level diffs between versions of your app.

**Comment by bmanderscheid** [comment]
Its too big, that's a shame. It's for an app in Nissan dealerships all across North America.   It has several 360 spins of all model, trims and colors utilizing sprite sheets, and edges around 1GB at the moment.  Hurry up and take our money! This would be perfect!

**Comment by bmanderscheid** [comment]
Do you foresee this being an option in the pricing tiers at some point?

**Comment by piyushjo** [comment]
Yes. We don't charge for the service today but we do want to make the service useful to as many apps as we can so expanding the allowable app size is definitely something we will consider.

**Comment by piyushjo** [comment]
Hey @bmanderscheid would you be open to a call with our CodePush team so we can understand your scenario and usage and see if we can figure out a workaround? Feel free to send me an email at piyushjo AT Microsoft.com and we can find some time. thank you.

**Comment by bmanderscheid** [comment]
Yes that'd be great!  I'll be in touch soon, thanks

**Comment by adriano-di-giovanni** [comment]
> Note that each time you perform a release, you'll need to upload the whole app, and the server will take care of generating file-level diffs between versions of your app.

@silhouettes,
I understand that CodePush handles deployments as follows:
* diffs deployment(i) with deployment(i-1)
* sends new and updated files to clients
* removes deleted files on clients

If so, updates are always incremental but we send the whole app.

Do you confirm?

Thanks

**Comment by silhouettes** [comment]
@adriano-di-giovanni - yes, everything you said is more or less correct. You always upload the whole app, and the update sent to clients is typically incremental. We also generate diffs for multiple versions beyond (i-1), but I can't be specific how many versions back because that may change.

**Comment by saifeer** [comment]
Hey @silhouettes, I have been trying to test the incremental updates for our app and it seems to not work for me. Like @bmanderscheid, we have an app that is quite big - around 300MB. I tried to first run the app, then change 1 file and push an update to code-push as v1 and then change the same file again and push another update to code-push as v2. 

I found that the app detected the updates correctly on both occasions. However, on both updates, the update.zip file that got downloaded contained the entire www directory contents. 

On both occasions, I used the command "appcenter codepush release-cordova -d Production" to generate the releases and on both occasions the console showed that the file change was picked up correctly:
`Merging and updating files from [www, platforms/ios/platform_www] to platforms/ios/www
  copy  www/index.html platforms/ios/www/index.html (updated file)`

Note that all of this testing is being done on a Simulator. Could this be a problem?
